arduino-ide-extension/src/browser/contributions/board-selection.ts

```diff
@@ -239,14 +240,18 @@ PID: ${PID}`;
       };
 
       // Board menu
-      const menuAction = { commandId: id, label: name };
+      const menuAction = {
+        commandId: id,
+        label: name,
+        order: `${index}`,
```

Comment:
We can check them together, but I think this is the classic alphabetic ordering issue. 

```sh
% node    
Welcome to Node.js v14.19.0.
Type ".help" for more information.
> ['1', '2', '3', '11', '12', '111'].sort()
[ '1', '11', '111', '12', '2', '3' ]
> .exit
```

With some zero padding, it works for me locally:

```diff
diff --git a/arduino-ide-extension/src/browser/contributions/board-selection.ts b/arduino-ide-extension/src/browser/contributions/board-selection.ts
index 2241c925..466e4f63 100644
--- a/arduino-ide-extension/src/browser/contributions/board-selection.ts
+++ b/arduino-ide-extension/src/browser/contributions/board-selection.ts
@@ -243,7 +243,7 @@ PID: ${PID}`;
       const menuAction = {
         commandId: id,
         label: name,
-        order: `${index}`,
+        order: String(index).padStart(4), // pads with leading zeros for alphanumeric sort where order is 1, 2, 11, and NOT 1, 11, 2
       };
       this.commandRegistry.registerCommand(command, handler);
       this.toDisposeBeforeMenuRebuild.push(
```

----

Great that your changes revealed the flaw in the codebase. These are probably all incorrect:

https://github.com/arduino/arduino-ide/blob/61a11a0857dd03c6057a558604fba810ac5955c0/arduino-ide-extension/src/browser/boards/boards-data-menu-updater.ts#L114

https://github.com/arduino/arduino-ide/blob/61a11a0857dd03c6057a558604fba810ac5955c0/arduino-ide-extension/src/browser/contributions/sketch-control.ts#L179

👇 This probably works only because the _open recent_ is limited to ten items.
https://github.com/arduino/arduino-ide/blob/61a11a0857dd03c6057a558604fba810ac5955c0/arduino-ide-extension/src/browser/contributions/open-recent-sketch.ts#L96

Reply:
Nice one! Thank you for the suggestion!

Reply:
I've pushed the suggestion, could you please verify it now?